spec/acceptance/device_spec.rb

```diff
@@ -63,6 +63,15 @@
         expect(status).to eq 0
       end
 
+      it 'outputs resources as yaml' do
+        stdout_str, status = Open3.capture2e("puppet resource #{common_args} device_provider --to_yaml")
+        expected_values = 'device_provider: |2\n\s+wibble:\n\s+ensure: :present\n\s+string: sample\n\s+string_ro: fixed\n\s+string_param: default value'
```

Comment:
@DavidS 

@DavidS It seems the YAML generation on older Puppet versions produces a slightly different output.

New:
```yaml
device_provider: |2
  wibble:
  ensure: :present
   string: sample
   string_ro: fixed
```

Old:
```yaml
device_provider:
  wibble:
  ensure: :present
   string: sample
   string_ro: fixed
```

I've tinkered with Rubular and managed to come up with a regexp covering both cases but that regexp fails during the test.

Regexp tested:

`device_provider:( \|2)?\n\s+wibble:\n\s+ensure: :present\n\s+string: sample\n\s+string_ro: fixed\n\s+string_param: default value`

Works in Rubular, fails in rspec.

Any suggestions appreciated!